aws/resource_aws_glue_job.go

```diff
@@ -0,0 +1,326 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/glue"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+)
+
+func resourceAwsGlueJob() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsGlueJobCreate,
+		Read:   resourceAwsGlueJobRead,
+		Update: resourceAwsGlueJobUpdate,
+		Delete: resourceAwsGlueJobDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"allocated_capacity": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Default:      10,
+				ValidateFunc: validation.IntBetween(2, 100),
+			},
+			"command": {
+				Type:     schema.TypeList,
+				Required: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Default:  "glueetl",
+						},
+						"script_location": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+					},
+				},
+			},
+			"connections": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"default_arguments": {
+				Type:     schema.TypeMap,
+				Optional: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"execution_property": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Computed: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"max_concurrent_runs": {
```

Comment:
Is there a reason to have this in an `execution_property` block property instead of having `max_concurrent_runs` at the top level like `max_retries` is?

Just to match the API? A potential for more options in `execution_property`?

Reply:
Just to match the API 😄 We've gotten bitten in the past by colliding attribute names. https://docs.aws.amazon.com/sdk-for-go/api/service/glue/#CreateJobInput

Reply:
Valid! Good thing to keep in mind then.